Assets/SEE/XR/RadialSelection.cs

```diff
@@ -0,0 +1,409 @@
+                GlobalActionHistory.Execute((ActionStateType)ActionStateTypes.AllRootTypes.AllElements().FirstOrDefault(a => a.Name == "Hide"));
+            }
+            if (actions[i] == "HideUnselected")
+            {
+                HideMode = HideModeSelector.HideUnselected;
+                GlobalActionHistory.Execute((ActionStateType)ActionStateTypes.AllRootTypes.AllElements().FirstOrDefault(a => a.Name == "Hide"));
+            }
+            if (actions[i] == "HideOutgoing")
+            {
+                HideMode = HideModeSelector.HideOutgoing;
+                GlobalActionHistory.Execute((ActionStateType)ActionStateTypes.AllRootTypes.AllElements().FirstOrDefault(a => a.Name == "Hide"));
+            }
+            if (actions[i] == "HideIncoming")
+            {
+                HideMode = HideModeSelector.HideIncoming;
+                GlobalActionHistory.Execute((ActionStateType)ActionStateTypes.AllRootTypes.AllElements().FirstOrDefault(a => a.Name == "Hide"));
+            }
+            if (actions[i] == "HideAllEdgesOfSelected")
+            {
+                HideMode = HideModeSelector.HideAllEdgesOfSelected;
+                GlobalActionHistory.Execute((ActionStateType)ActionStateTypes.AllRootTypes.AllElements().FirstOrDefault(a => a.Name == "Hide"));
+            }
+            if (actions[i] == "HideForwardTransitiveClosure")
+            {
+                HideMode = HideModeSelector.HideForwardTransitiveClosure;
+                GlobalActionHistory.Execute((ActionStateType)ActionStateTypes.AllRootTypes.AllElements().FirstOrDefault(a => a.Name == "Hide"));
+            }
+            if (actions[i] == "HideBackwardTransitiveClosure")
+            {
+                HideMode = HideModeSelector.HideBackwardTransitiveClosure;
+                GlobalActionHistory.Execute((ActionStateType)ActionStateTypes.AllRootTypes.AllElements().FirstOrDefault(a => a.Name == "Hide"));
+            }
+            if (actions[i] == "HideAllTransitiveClosure")
+            {
+                HideMode = HideModeSelector.HideAllTransitiveClosure;
+                GlobalActionHistory.Execute((ActionStateType)ActionStateTypes.AllRootTypes.AllElements().FirstOrDefault(a => a.Name == "Hide"));
+            }
+            if (actions[i] == "HighlightEdges")
+            {
+                HideMode = HideModeSelector.HighlightEdges;
+                GlobalActionHistory.Execute((ActionStateType)ActionStateTypes.AllRootTypes.AllElements().FirstOrDefault(a => a.Name == "Hide"));
+            }
+            else
+            {
+                GlobalActionHistory.Execute((ActionStateType)ActionStateTypes.AllRootTypes.AllElements().FirstOrDefault(a => a.Name == actions[i]));
+            }
+        }
+        else
+        {
+            if (subMenus[i].Item2 == null)
+            {
+                if (subMenus[i].Item1 == "Hide")
+                {
+                    numberOfRadialPart = 11;
```

Comment:
This magic number should be replaced with a constant. Or, even better, the "hide" actions could be put into a static list, then we can just refer to its count here.